src/vscode-bicep/src/tree/AzLoginTreeItem.ts

```diff
@@ -0,0 +1,175 @@
+    let azureAccount: AzureAccountResult = await this._azureAccountTask;
+    if (typeof azureAccount === "string") {
+      // Refresh the AzureAccount, to handle Azure account extension installation after the previous refresh
+      this._azureAccountTask = this.loadAzureAccount(this._testAccount);
+      azureAccount = await this._azureAccountTask;
+    }
+
+    if (typeof azureAccount === "string") {
+      const label: string =
+        azureAccount === "notInstalled"
+          ? localize(
+              "installAzureAccount",
+              "$(warning) Install Azure Account Extension and try deploy again"
+            )
+          : localize(
+              "updateAzureAccount",
+              '$(warning) Update Azure Account Extension to at least version "{0}"',
+              minAccountExtensionVersion
+            );
+      const result: AzExtTreeItem = new GenericTreeItem(this, {
+        label,
+        commandId: extensionOpenCommand,
+        contextValue: "azureAccount" + azureAccount,
+        includeInTreeItemPicker: true,
+      });
+      result.commandArgs = [azureAccountExtensionId];
+      return [result];
+    }
+
+    const contextValue = "azureCommand";
+
+    if (azureAccount.status != "LoggedIn") {
+      return [
+        new GenericTreeItem(this, {
+          label: "$(sign-in) " + signInLabel,
+          commandId: signInCommandId,
+          contextValue,
+          id: signInCommandId,
+          includeInTreeItemPicker: true,
+        }),
+        new GenericTreeItem(this, {
+          label: "$(add) " + createAccountLabel,
+          commandId: createAccountCommandId,
+          contextValue,
+          id: createAccountCommandId,
+          includeInTreeItemPicker: true,
+        }),
+      ];
+    }
+
+    return [new GenericAzExtTreeItem(this, "", "")];
+  }
+
+  public async getIsLoggedIn(): Promise<boolean> {
+    const azureAccount: AzureAccountResult = await this._azureAccountTask;
+    return (
+      typeof azureAccount !== "string" && azureAccount.status === "LoggedIn"
+    );
+  }
+
+  public compareChildrenImpl(
+    item1: AzExtTreeItem,
+    item2: AzExtTreeItem
+  ): number {
+    if (item1 instanceof GenericTreeItem && item2 instanceof GenericTreeItem) {
+      return 0; // already sorted
+    } else {
+      return super.compareChildrenImpl(item1, item2);
+    }
+  }
+
+  private async loadAzureAccount(
+    azureAccount: AzureAccount | undefined
+  ): Promise<AzureAccountResult> {
+    if (!azureAccount) {
+      const extension: Extension<AzureAccount> | undefined =
+        extensions.getExtension<AzureAccount>(azureAccountExtensionId);
+      if (extension) {
+        try {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          if (
+            semver.lt(extension.packageJSON.version, minAccountExtensionVersion)
+          ) {
+            return "needsUpdate";
+          }
+        } catch {
```

Comment:
This shouldn't ever throw, right?  If so, probably better to not try/catch, and let users and telemetry see it if it fails.

Reply:
Agreed. This one was inspired from vscode-azuretools code. I think they have it there. It's only comparing versions. So removed try/catch here:
https://github.com/Azure/bicep/commit/a4f905e66fdf2d01b8646f3634a01bdf59f2f80a

Thanks!